ganja.js

```diff
@@ -646,24 +646,45 @@
             // Numbers change the current color.
               if (typeof o =='number') { color='#'+(o+(1<<25)).toString(16).slice(-6); return ''; };
             // All other elements are rendered ..
-              var b1=o.Grade(1).VLength>0.001,b2=o.Grade(2).VLength>0.001,b3=o.Grade(3).VLength>0.001;
-            // Points
-              if (b1 && !b2 && !b3) {
-                o.set(o.Scale(-1/o.Dot(cga2d_ni).s));
+              var einf_part = o.Dot(cga2d_no.Scale(-1));  // O_i + n_o O_oi
+              var eo_part = cga2d_ni.Scale(-1).Dot(o);  // O_o + O_oi n_i
+              if (einf_part * 1e-9 > eo_part) {
+                // direction or dual - nothing to render
+                return "";
+              }
```

Comment:
Hmm, this condition needs some though - it's described as `-ni | x == 0` in GA4CS, but that doesn't work in the presence of epsilons...

Reply:
ok I'll fix the syntax problem here for now (needs a scalar selector) - that will probably also resolve the epsilon issues .. inside ganja there's no operator overloading so now line 651 is just comparing NaN to an mv always evaluating to false.